tasks.py

```diff
@@ -231,6 +233,60 @@ def dev_env(ctx, architecture="amd64", name="kind", cni=None):
         tmp.flush()
         run("kubectl apply -f {}".format(tmp.name), echo=True)
 
+    if protocol == "bgp":
+        print("Configuring MetalLB with a BGP test environment")
+        bgp_dev_env()
+    else:
+        print("Leaving MetalLB unconfigured")
+
+
+# Configure MetalLB in the dev-env for BGP testing.  Start an frr based BGP
+# router in a container and configure MetalLB to peer with it.
+# See dev-env/bgp/README.md for some more information.
+def bgp_dev_env():
+    dev_env_dir = os.getcwd() + "/dev-env/bgp"
+
+    # TODO -- The IP address handling will need updates to add support for IPv6
+
+    node1_ip = run('docker inspect -f "{{ '
+            '.NetworkSettings.Networks.kind.IPAddress }}" kind-control-plane',
+            echo=True)
+    node2_ip = run('docker inspect -f "{{ '
+            '.NetworkSettings.Networks.kind.IPAddress }}" kind-worker',
+            echo=True)
+    node3_ip = run('docker inspect -f "{{ '
+            '.NetworkSettings.Networks.kind.IPAddress }}" kind-worker2',
+            echo=True)
```

Comment:
All of this fail locally here :-(

I'm using kind: 
```
$ kind version
kind v0.7.0 go1.13.6 linux/amd64
```
Are you using a different version of kind?


The output to help to see why it fails here is:
```
$  docker inspect  kind-control-plane
[
   ....
    {
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": ...,
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "6443/tcp": [
                    {
                        "HostIp": "127.0.0.1",
                        "HostPort": "32768"
                    }
                ]
            },
            "SandboxKey": "/var/run/docker/netns/bf9ddfad6a78",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": ...
            "Gateway": ...
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": ...
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "MacAddress":  ....
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "596f61bc70fe5d0960da560b55154965274a784dce236a17bd740e8cc4ca6b39",
                    "EndpointID": "e80728eb8b5ce97a017a07912bcf2dfe6181c63e11f16bcc0d3982587ce7c39e",
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.4",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:11:00:04",
                    "DriverOpts": null
                }
            }
        }
    }
]
$ docker inspect -f "{{ .NetworkSettings.Networks.kind.IPAddress }}" kind-control-plane
<no value>
$ kubectl get nodes
NAME                 STATUS   ROLES    AGE     VERSION
kind-control-plane   Ready    master   2m19s   v1.17.0
kind-worker          Ready    <none>   103s    v1.17.0
kind-worker2         Ready    <none>   103s    v1.17.0
(metallb-venv) rodrigo@lindsay: ~/src/kinvolk/metallb :bgptest$ docker inspect -f "{{ .NetworkSettings.Networks.kind.IPAddress }}" kind-control-plane
<no value>

$ kubectl get nodes -o wide
NAME                 STATUS   ROLES    AGE   VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE       KERNEL-VERSION   CONTAINER-RUNTIME
kind-control-plane   Ready    master   14m   v1.17.0   172.17.0.4    <none>        Ubuntu 19.10   5.7.0-2-amd64    containerd://1.3.2
kind-worker          Ready    <none>   13m   v1.17.0   172.17.0.2    <none>        Ubuntu 19.10   5.7.0-2-amd64    containerd://1.3.2
kind-worker2         Ready    <none>   13m   v1.17.0   172.17.0.3    <none>        Ubuntu 19.10   5.7.0-2-amd64    containerd://1.3.2

```

Maybe we can use kubectl internal-ip field to get the IPs?

Reply:
kind since 0.8.0+ changed the network model and uses a custom bridge named kind

Reply:
iterating over the range works for both versions, this snippet is from the kind code and works for ipv4 and ipv6, but just remove the family that you don't want :)
```
	// retrieve the IP address of the node using docker inspect
	cmd := exec.Command("docker", "inspect",
		"-f", "{{range .NetworkSettings.Networks}}{{.IPAddress}},{{.GlobalIPv6Address}}{{end}}",
		n.name, // ... against the "node" container
	)
```

Reply:
Thanks!

I _hope_ kubectl might be simpler but both will work :)

Reply:
`kubectl` works fine for getting the Node IPs, but I still have to use `docker inspect` to get the IP of the `frr` container (BGP router container)

Reply:
I've addressed this using `kubectl` for Node IPs, and an updated `docker inspect` for the `frr` container.  It still works for me with kind 0.8.1, but hopefully someone else can test with older kind.

Reply:
... if we actually care about older kind?  We could also just add a version check on kind so we don't have to deal with supporting differences between versions.  Let's see if it works first, though.

Reply:
No, sorry, of course it's totally okay to upgrade kind. I just wanted to avoid "it only works on Rusell's setup" thing :-)

Will update kind and try again :)